sdk/search/azure-search/src/main/java/com/azure/search/models/SuggestRequest.java

```diff
@@ -0,0 +1,384 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is
+// regenerated.
+
+package com.azure.search.models;
+
+import com.azure.core.annotation.Fluent;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * Parameters for filtering, sorting, fuzzy matching, and other suggestions
+ * query behaviors.
+ */
+@Fluent
+public final class SuggestRequest {
```

Comment:
@navalev From what I can tell, this class is only used internally. Can it be package-scope?

Reply:
Same question for `AutocompleteRequest`.

Reply:
@brjohnstmsft since this a generated class, all changes should be done in the swagger config and then re-generate the code. 
But in this case, the result classes are used outside of the model classes by DocumentsImpl so can't be private

Reply:
@navalev It seems that `DocumentsImpl` is public, when it should be package-scope. If it were, then I believe those models could be package-scope as well. The real public interface is in `SearchIndexClient` and `SearchIndexAsyncClient`.

Is my understanding correct?

If so, we'll adjust the codgen config as needed after this PR is merged.

Reply:
This will require a review on the whole access modifiers in the generated code, and yes can be done seperatly.